src/TLuaInterpreter.cpp

```diff
@@ -1556,17 +1556,27 @@ int TLuaInterpreter::getLineCount(lua_State* L)
 // Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#getColumnNumber
 int TLuaInterpreter::getColumnNumber(lua_State* L)
 {
+    QString windowName = QLatin1String("main");
     Host& host = getHostFromLua(L);
     if (lua_isstring(L, 1)) {
-        string window = lua_tostring(L, 1);
-        QString _window = window.c_str();
-        lua_pushnumber(L, mudlet::self()->getColumnNumber(&host, _window));
+        if (!lua_isstring(L, 1)){
+            lua_pushfstring(L, "getColumnNumber: bad argument #1 type (window name as string expected, got %s!)", luaL_typename(L, 1));
+            return lua_error(L);
+        }
+        int result = mudlet::self()->getColumnNumber(&host, windowName);
+        if (result == -1) {
+            lua_pushnil(L);
+            lua_pushfstring(L, "window \"%s\" does not exist", windowName.toUtf8().constData());
+            return 2;
+        } else {
+            lua_pushnumber(L, result);
+            return 1;
+        }
         return 1;
     } else {
-        lua_pushnumber(L, host.mpConsole->getColumnNumber());
+        lua_pushnumber(L, mudlet::self()->getColumnNumber(&host, windowName));
```

Comment:
This line needs to be ***reverted*** as well. I.e. it should be:
```C++
         lua_pushnumber(L, host.mpConsole->getColumnNumber()); 
```

Sorry for any confusion - but I have been working very recently on code that *does* merge both types of call to a `getColumnNumber(`...`)` to be a single call to a `TConsole` method even when `windowName` is empty or `main` so didn't spot that it will not work on the underlying code that is here right now! :blush:

Reply:
applied.